tests/RPM/TagsSpec.hs

```diff
@@ -0,0 +1,26 @@
+module RPM.TagsSpec (main, spec) where
+
+import Test.Hspec
+import RPM.Tags(mkTag, Tag(..))
+import qualified Data.ByteString as BS
+
+main :: IO ()
+main = hspec spec
+
+spec :: Spec
+spec = describe "RPM.Tags" $
+  describe "mkTag" $ do
+    it "returns Nothing for unknown `tag' value" $ do
+      let store = BS.pack [0,0,0,5]
+      let tag = mkTag store 9999 0 0 0
+      tag `shouldBe` Nothing
+
+    it "returns FileModes for 1030 `tag' value" $ do
+      let store = BS.pack [0,0,0,5]
+      let tag = mkTag store 1030 3 0 2
+      tag `shouldBe` Just (FileModes [0,5])
+
+    it "returns Nothing for 1030 `tag' value and `ty' != 3" $ do
+      let store = BS.pack [0,0,0,5]
+      let tag = mkTag store 1030 0 0 2
+      tag `shouldBe` Nothing
```

Comment:
I think you are on the right track here.  Certainly we don't want tests for every single tag, but it'd be worth testing each type of tag and their corner cases (empty lists, count is 0, that kind of thing).